src/main/scala/it/pagopa/interop/authorizationprocess/service/impl/CatalogManagementServiceImpl.scala

```diff
@@ -1,29 +1,21 @@
 package it.pagopa.interop.authorizationprocess.service.impl
 
-import it.pagopa.interop.authorizationprocess.service.{CatalogManagementInvoker, CatalogManagementService}
-import it.pagopa.interop.catalogmanagement.client.api.EServiceApi
-import it.pagopa.interop.catalogmanagement.client.invoker.BearerToken
-import it.pagopa.interop.catalogmanagement.client.model.EService
-import com.typesafe.scalalogging.{Logger, LoggerTakingImplicit}
-import it.pagopa.interop.commons.logging.{CanLogContextFields, ContextFieldsToLog}
-import it.pagopa.interop.commons.utils.withHeaders
+import it.pagopa.interop.authorizationprocess.service.CatalogManagementService
 
-import java.util.UUID
-import scala.concurrent.Future
-import it.pagopa.interop.catalogmanagement.client.invoker.ApiRequest
+import it.pagopa.interop.catalogmanagement.model.CatalogItem
+import it.pagopa.interop.authorizationprocess.common.readmodel.ReadModelCatalogQueries
+import it.pagopa.interop.commons.cqrs.service.ReadModelService
+import it.pagopa.interop.commons.utils.TypeConversions._
+import it.pagopa.interop.authorizationprocess.error.AuthorizationProcessErrors.EServiceNotFound
 
-final case class CatalogManagementServiceImpl(invoker: CatalogManagementInvoker, api: EServiceApi)
-    extends CatalogManagementService {
+import java.util.UUID
+import scala.concurrent.{Future, ExecutionContext}
 
-  implicit val logger: LoggerTakingImplicit[ContextFieldsToLog] =
-    Logger.takingImplicit[ContextFieldsToLog](this.getClass)
+final object CatalogManagementServiceImpl extends CatalogManagementService {
```

Comment:
`final` not required for objects
```suggestion
object CatalogManagementServiceImpl extends CatalogManagementService {
```